### PR TITLE
1.6 Port: Dynamic Versioning

### PR DESCRIPTION
**Pull request by mruberry**
Port of https://github.com/pytorch/pytorch/pull/40279.

Addresses https://github.com/pytorch/pytorch/issues/40140.

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

**Comment by seemethere** [comment]
Will merge, pending approval from @zdevito

**Review by zdevito — APPROVED** [review]

**Pull request merged**
